crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -816,6 +816,20 @@ where
                     flake8_pie::rules::non_unique_enums(self, stmt, body);
                 }
 
+                if ruff::rules::is_dataclass(self, decorator_list) {
```

Comment:
Marginal, but can we check if either of the relevant rules are enabled before doing this `is_dataclass` check, since this'll be more expensive (requires resolving all the decorators)? E.g.:

```rust
if self.settings.rules.any_enabled(...) {
  if ruff:rules::is_dataclass(...) {
    ...
  }
}
```

Reply:
Done